### DISCUSSION TITLE
Certbot fails during installation (install.sh)

### DISCUSSION THREAD

**Discussion by isential-gmbh** [rustdesk/rustdesk-server-pro#209] [Q&A]
Hello,
 
I am installing RustDesk on a current Ubuntu server. I am referring to this server as the RustDesk server. There are no domains set up on this server; I only have a static IP address.
 
On another server, I have set up a DNS forwarding (A record) pointing to the IP address of the RustDesk server:
 
rustdesk.example.com = IP address of the RustDesk server
 
I am performing the installation using the following command:
 
bash <(wget -qO- https://raw.githubusercontent.com/rustdesk/rustdesk-server-pro/main/install.sh)
 
During the installation, I select "DNS = Setup Rustdesk with TLS and your own domain" and specify the address as rustdesk.example.com.
 
When prompted with:
 
"Do you want to install Certbot with snap?"
 
I answer “Yes”.
 
However, I receive the following error message:
 
"Sorry, the TLS certificate for rustdesk.example.com failed to generate!
Please check that port 80/443 are correctly port forwarded, and that the DNS record points to this server's IP."
 
In the file /etc/ufw/applications.d/rustdesk-server, I have the following content:
 
[RustDesk]
title=RustDesk Remote Desktop
description=Remote Desktop application
ports=80,443/tcp
 
Additionally, I have activated the firewall rule:
 
ufw allow RustDesk
 
and executed the following commands:
 
ufw allow 21115:21119/tcp
ufw allow 8000/tcp
ufw allow 21116/udp
sudo ufw enable
 
How do I properly install the certificate for "rustdesk.example.com" with Certbot?

**Comment by rustdesk** [comment]
@dinger1986 could you help on this?

**Reply by dinger1986** [reply]
Make sure all the ports are open and the dns resolves correctly, https://www.yougetsignal.com/tools/open-ports/ 

Other than that I dont know cause it was all rewritten by someone and totally changed how it works and the logic, it looks over complex.
 
I have my original script which I can share which works every time with no issues.

**Reply by isential-gmbh** [reply]
Thank you! The ports are open.

I'm certainly no networking expert and I'm struggling with this. I just thought I'd set up an A record on the other server pointing to its IP address.

**Comment by rustdesk** [comment]
> On another server, I have set up a DNS forwarding (A record) pointing to the IP address of the RustDesk server

Since your dns on another server, I think you should answer `No` to  `"Do you want to install Certbot with snap?"`.

**Reply by isential-gmbh** [reply]
Okay, thank you. But how is the traffic secured then?

**Reply by rustdesk** [reply]
Your `another server` can secure it.

**Reply by rustdesk** [reply]
BTW, Certbot secure the http traffic for the web console only. Our RustDesk connections are secured by our p2p encryption, nothing to do the certbot here.

**Comment by ZetaWaves** [comment]
#1 You need to have a /snap snap symlink.
#2 You need to install snap and certbot.
